proto/substrait/algebra.proto

```diff
@@ -1164,4 +1163,10 @@ message AggregateFunction {
     // Use only distinct values in the aggregation calculation.
     AGGREGATION_INVOCATION_DISTINCT = 2;
   }
+
+  // This rel is used  to create references,
+  // in case we refer to a RelRoot field names will be ignored
```

Comment:
Would referring to RelRoot make sense?  Wouldn't that lead to a circular tree?

Reply:
IIRC Jacques told me at some point that in the original design there is only supposed to be one `RelRoot`, and that it should indeed be the root of a relation, while the normal `Rel`s in `PlanRel` were exclusively intended to be used for this referencing logic. @jacques-n please correct me if I'm misremembering or if I misunderstood.

Personally I don't like this at all; IMO this structure should have been reflected in the protobuf message structure rather than making a repeated oneof, i.e. something like `PlanRel { RootRel root, repeated Rel subtrees }`. But that's just my opinion, and it's a bit late for that.

Since none of the above was ever documented AFAICT, and since rel references didn't exist at all until this proposal, I think most readers (including @curino I think) would interpret plans the way Arrow's consumer does it now and the way I initially interpreted it, being:

 - a `Plan` actually consists of one or more plans;
 - the only functional difference between `RootRel` and a normal `Rel` is that the former attaches column names and the other does not.

in which case it'd be perfectly fine to refer to a `RelRoot` as well.

IMO the better way to avoid circular references and to make parsing the plans easier is to require that you can only refer to relation trees with an index < the one you're referring from. Thus far everything in Substrait could be parsed with a single depth-first tree traversal (with carefully chosen field order); without this requirement this would no longer be the case.